modules/apps/web-experience/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/util/AssetPublisherUtil.java

```diff
@@ -358,23 +358,44 @@ public static String filterAssetTagNames(
 			int end)
 		throws Exception {
 
+		List<AssetEntry> assetEntries = null;
+
 		if (isSearchWithIndex(portletName, assetEntryQuery)) {
-			return _assetHelper.searchAssetEntries(
-				assetEntryQuery, getAssetCategoryIds(portletPreferences),
-				getAssetTagNames(portletPreferences), attributes, companyId,
-				assetEntryQuery.getKeywords(), layout, locale, scopeGroupId,
-				timeZone, userId, start, end);
+			BaseModelSearchResult<AssetEntry> baseModelSearchResult =
+				_assetHelper.searchAssetEntries(
+					assetEntryQuery, getAssetCategoryIds(portletPreferences),
+					getAssetTagNames(portletPreferences), attributes, companyId,
+					assetEntryQuery.getKeywords(), layout, locale, scopeGroupId,
+					timeZone, userId, start, end);
+
+			assetEntries = baseModelSearchResult.getBaseModels();
+		}
+		else {
+			assetEntryQuery.setEnd(end);
+			assetEntryQuery.setStart(start);
+
+			assetEntries = _assetEntryService.getEntries(assetEntryQuery);
 		}
 
-		int total = _assetEntryService.getEntriesCount(assetEntryQuery);
+		List<AssetEntry> results = new ArrayList<>();
+
+		for (AssetEntry assetEntry : assetEntries) {
+			AssetRendererFactory<?> assetRendererFactory =
+				AssetRendererFactoryRegistryUtil.
+					getAssetRendererFactoryByClassName(
+						assetEntry.getClassName());
 
-		assetEntryQuery.setEnd(end);
-		assetEntryQuery.setStart(start);
+			AssetRenderer<?> assetRenderer =
+				assetRendererFactory.getAssetRenderer(
+					assetEntry.getClassPK(),
+					AssetRendererFactory.TYPE_LATEST_APPROVED);
 
-		List<AssetEntry> results = _assetEntryService.getEntries(
-			assetEntryQuery);
+			if (assetRenderer.isDisplayable()) {
+				results.add(assetEntry);
+			}
+		}
 
-		return new BaseModelSearchResult<>(results, total);
+		return new BaseModelSearchResult<>(results, results.size());
```

Comment:
Usage of `results.size()` here will always give us `(end - start - nonDisplayable)` number of entries, but not the total count as it was before.